lib/connection.cpp

```diff
@@ -133,6 +133,8 @@ Connection::Connection(const QUrl& server, QObject* parent)
     , d(std::make_unique<Private>(std::make_unique<ConnectionData>(server)))
 {
     d->q = this; // All d initialization should occur before this line
+    // sync loop:
+    connect(this, &Connection::syncDone, this, &Connection::getNewEvents);
```

Comment:
There's a caveat here: `syncDone` is only invoked if `sync()` finished successfully. Think how you're going to handle sync errors (hint: that's one reason I'd prefer to keep the two methods apart; hint2: take a look at `Connection::sync()` implementation and consider how a client would/should behave in case of `SyncJob::failure`).